public/models/aeon_record_mapper.rb

```diff
@@ -8,6 +8,26 @@ def initialize(record)
         @record = record
     end
 
+    def self.mapper_for(record)
```

Comment:
I appreciate the push to improve the logic of building the mapper, but I don't believe that this is the way to do it. References to the subclasses of a base class should not be included in that base class.

If you want to move this logic out of the Ruby partial and into a class, please create an AeonRecordMapperFactory and define the method there.

Reply:
I've pushed a commit that changes the approach to this. Rather than using config, it allows the mapper classes to register for record types that they can handle.

I think this is cleaner and gets the reference to subclass names out of the super class.

It is also an ArchivesSpacey way of doing things. See for example this pattern used in job_runner.rb.